src/lib.rs

```diff
@@ -14,6 +14,79 @@
 
 pub use showata::Showable;
 
+use base64;
+use data_url::DataUrl;
+use fantoccini::{Client, Locator};
+use retry::{delay::Exponential, retry_with_index, OperationResult};
+//use futures_retry::{RetryPolicy, StreamRetryExt};
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+use std::process::Command;
+
+/// Spawn in background a webdriver, currently support is limited to
+/// geckodriver. Please see [geckodriver doc](https://github.com/mozilla/geckodriver) and install it.
+/// TODO: allow [chromium webdriver](https://chromedriver.chromium.org/) and update documentation
+pub fn spawn_webdriver(
+    webdriver_name: &str,
+    port: Option<u64>,
+) -> Result<(std::process::Child, u64), Box<dyn std::error::Error>> {
+    let mut try_port = match port {
+        Some(n) => n,
+        None => 4444,
+    };
+    let webdriver_process = retry_with_index(Exponential::from_millis(100), |current_try| {
+        if current_try > 3 {
+            return OperationResult::Err("did not succeed within 3 tries");
+        }
+        try_port += current_try;
+        let try_command = Command::new(webdriver_name)
+            .args(&["--port", &try_port.to_string()])
+            .spawn();
+        match try_command {
+            Ok(cmd) => OperationResult::Ok(cmd),
+            Err(_) => OperationResult::Retry("Trying with another port"),
+        }
+    })
+    .unwrap();
+    Ok((webdriver_process, try_port))
+}
+
+/// Create a headless browser instance.
+/// Code from : https://github.com/jonhoo/fantoccini/blob/master/tests/common.rs
+/// The chrome case will be commented for now and will be tested later.str
+/// It also need the port from the webdriver.
+pub async fn create_headless_client(
+    client_type: &str,
```

Comment:
could this be an enum that would force user to specify a supported browser instead of a free string?